### PR TITLE
Feat/ade 495/table filters

### PR DESCRIPTION
**Pull request by SamMurphyDev**

### CODE REVIEW

File: src/main/resources/js/TableFilter.js [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+ *
+ * @param {HTMLElement} element Item from dropdown to add
+ * @param {string} searchField Id of search field
+ * @param {string} list Id of dropdown list
+ */
+const addTagToMultiSelect = (searchField, list, element) => {
+  if (element !== undefined) {
+    $(`#${searchField}`).before(
+        `<span class='tag'>${$(
+            element).text()}<button type="button" class="close">&#x2715</button></span>`);
```
Comment (shouvSarker) [inline src/main/resources/js/TableFilter.js:131]: Is this for the `X` symbol? https://www.codetable.net/hex/2715

Can we get this as a constant from somewhere / set as a constant?

File: src/main/resources/js/TableFilter.js [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+      tempList.sort().forEach((value) => itemLists[listRef].add(value));
+
+      $(`#${listRef}`).append(
+          ...generateHTMLTagsForListItems(itemLists[listRef]));
+    });
+  })([{key: 'technology', listRef: appConstants.TECHNOLOGIES_LIST},
+    {key: 'status', listRef: appConstants.STATUS_LIST}]);
+
+  /*
+   * Standard text search
```
Comment (shouvSarker) [inline src/main/resources/js/TableFilter.js:266]: Could we have some more details about this search in the comment? i.e. I notice we are doing substring with 7 on target id.

File: src/main/resources/js/TableFilter.js [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+  /*
+   * MultiSelect Type Ahead Event handlers
+   */
+  $(`.searchable`).on({
+    'input': ({target}) => {
+      setHiddenFlagsOnDropdownList(searchToList[target.id],
+          $(target).val().toLowerCase());
+    },
+    'keyup': ({keyCode, target}) => {
+      if (keyCode === 13 && $(target).val() != '') {
```
Comment (shouvSarker) [inline src/main/resources/js/TableFilter.js:288]: Does `keyCode` 13 mean the `Enter` key?  Could we have this in a const or have as a constant from somewhere?

File: src/main/resources/vm/IndexPage.vm [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
 #* @vtlvariable name="rows" type="java.lang.reflect.Array" *#
 <p class="add-button">
-  #if($blueprint.blueprintId != "")
-    <a class="aui-button create-from-template-button conf-macro output-inline"
-       data-space-key="$blueprint.spaceKey"
-       href="$blueprint.baseUrl?createDialogSpaceKey=$blueprint.spaceKey&createDialogBlueprintId=$blueprint.blueprintId"
-       data-content-blueprint-id="$blueprint.blueprintId" data-hasbody="false"
-       data-macro-name="create-from-template">Add Fedex Idea</a>
-  #end
+    #if($blueprint.blueprintId != "")
```
Comment (shouvSarker) [inline src/main/resources/vm/IndexPage.vm:5]: Change in indentation seems to be the only difference? Just double checking.

File: src/main/resources/vm/IndexPage.vm [hunk 5]
```diff
@@ -21,15 +21,56 @@
...
+  <tbody class="table-content">
       #foreach($row in $rows)
       <tr>
         <td class="cell-title">$row.title</td>
         <td class="cell-description">$row.description.getRenderedValue()</td>
         <td class="cell-technology">$row.technologies.getRenderedValue()</td>
         <td class="cell-status">$row.status.getRenderedValue()</td>
         <td class="cell-owner">
-          #if($row.owner.getRenderedValue())
+            #if($row.owner.getRenderedValue())
```
Comment (shouvSarker) [inline src/main/resources/vm/IndexPage.vm:73]: Should the line below `if` be indented?

File: src/main/resources/vm/IndexPage.vm [hunk 6]
```diff
@@ -21,15 +21,56 @@
...
+        <div id="technologies-list-container" class="list-container">
+          <ul id="technologies-list" class="select-list hidden" role="listbox">
+          </ul>
+        </div>
+      </div>
+    </td>
+    <td class="w-8">
+      <div id="status-multiselect-container" class="multiselect-container">
+        <div class="tags-input search-box">
+          <input id="search-status" class="searchable" autocomplete="chrome-off" tabindex="0"
```
Comment (shouvSarker) [inline src/main/resources/vm/IndexPage.vm:52]: Could we have a comment here linking to why we use `autocomplete="chrome-off"`?

File: src/main/resources/css/IndexTable.css [hunk 7]
```diff
@@ -7,16 +7,12 @@ tr {
...
-
-tr:last-child {
-  max-width: 10%;
+.table-content td {
+  padding: 0.25rem 1rem;
 }
 
-td {
-  padding: 0.25rem 1rem;
+tbody:nth-child(1) > tr:last-child, thead > tr:first-child {
```
Comment (shouvSarker) [inline src/main/resources/css/IndexTable.css:14]: Running prettier seems to make minor changes to this file and the other ones as well. Have another prettier run for a sanity check?

File: src/main/resources/js/TableFilter.js (hunk 1 above)
Comment (SamMurphyDev) [reply src/main/resources/js/TableFilter.js]: Moved to const. Javascript has no standard definition for the ASCII table

File: src/main/resources/js/TableFilter.js (hunk 2 above)
Comment (SamMurphyDev) [reply src/main/resources/js/TableFilter.js]: Added additional information to comment

File: src/main/resources/js/TableFilter.js (hunk 1 above)
Comment (SamMurphyDev) [reply src/main/resources/js/TableFilter.js]: `substring` now references a string length as it's stripping out a prefix. The appConstants have been changed to a template literal when required to incorporate the prefix.

File: src/main/resources/js/TableFilter.js (hunk 2 above)
Comment (SamMurphyDev) [reply src/main/resources/js/TableFilter.js]: `substring` now references a string length as it's stripping out a prefix. The appConstants have been changed to a template literal when required to incorporate the prefix.

File: src/main/resources/js/TableFilter.js (hunk 3 above)
Comment (SamMurphyDev) [reply src/main/resources/js/TableFilter.js]: Move the 13 to a const. JavaScript has no standard definition for keycodes

File: src/main/resources/vm/IndexPage.vm (hunk 4 above)
Comment (SamMurphyDev) [reply src/main/resources/vm/IndexPage.vm:5]: Fixed indentation

File: src/main/resources/vm/IndexPage.vm (hunk 6 above)
Comment (SamMurphyDev) [reply src/main/resources/vm/IndexPage.vm:54]: Put comment in linking to chromium bug tracker

File: src/main/resources/css/IndexTable.css (hunk 7 above)
Comment (SamMurphyDev) [reply src/main/resources/css/IndexTable.css]: Ran prettier

File: src/main/resources/js/TableFilter.js (hunk 1 above)
Comment (shouvSarker) [reply src/main/resources/js/TableFilter.js]: Nice!

File: src/main/resources/css/IndexTable.css (hunk 7 above)
Comment (shouvSarker) [reply src/main/resources/css/IndexTable.css]: Great! Nicely indented now.

**Review by shouvSarker — APPROVED** [review]
Looks great! Nice job Sam.

**Review by robinMcA — APPROVED** [review]

**Pull request merged**